### PR TITLE
fix(theme): litho UI elements recolor

### PR DESCRIPTION
**Pull request by OxrxL**

### CODE REVIEW

**Comment by oSumAtrIX** [comment]
Does this implementation consider theming the light mode theme?

**Comment by OxrxL** [comment]
> Does this implementation consider theming the light mode theme?

Already included inside integration java class.

File: app/src/main/java/app/revanced/integrations/patches/LithoThemePatch.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package app.revanced.integrations.patches;
+
+import app.revanced.integrations.utils.ThemeHelper;
+
+public class LithoThemePatch {
+    //Used by app.revanced.patches.youtube.layout.theme.patch.LithoThemePatch
+    public static int applyLithoTheme(int value) {
+        int newValue = 0;
+
+        if (!ThemeHelper.isDarkTheme()) {
```
Comment (OxrxL) [inline app/src/main/java/app/revanced/integrations/patches/LithoThemePatch.java:10]: This is what you're looking for @oSumAtrIX

**Comment by oSumAtrIX** [comment]
The current implementation is a bit unclear, please give the parameters and variables more explanatory names as well as comment your code to explain what it does for future reference.

**Comment by OxrxL** [comment]
> The current implementation is a bit unclear, please give the parameters and variables more explanatory names as well as comment your code to explain what it does for future reference.

Do the same thing of the hard codes patch:

Really easy:

This method check if the theme is white or dark and, based on this condition, isolates (with another condition) the individual color values from the register field p1, to subsequently change them to zero (transparent color) and hide the panel elements drawn by litho.

**Comment by oSumAtrIX** [comment]
> with another condition

This is unclear, please refactor the code as mentioned.

**Comment by OxrxL** [comment]
> > with another condition
> 
> This is unclear, please refactor the code as mentioned.

Are you blind? I already added the check for light theme. 🤷‍♂️

**Comment by oSumAtrIX** [comment]
I asked to refactor the code by properly naming your variables and parameters, not change the behavior.

**Comment by OxrxL** [comment]
> I asked to refactor the code by properly naming your variables and parameters, not change the behavior.

Yeah...sorry. There was a misunderstanding by me.

File: app/src/main/java/app/revanced/integrations/patches/LithoThemePatch.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import app.revanced.integrations.utils.ThemeHelper;
+
+public class LithoThemePatch {
+    // color constants used in relation with litho components
+    private static final int[] WHITECONSTANTS = {
+        -1, // comments chip background
+        -394759, // music related results panel background
+        -83886081, // video chapters list background
+    };
+    private static final int[] DARKCONSTANTS = {
```
Comment (oSumAtrIX) [inline app/src/main/java/app/revanced/integrations/patches/LithoThemePatch.java:14]:
```suggestion

    private static final int[] DARKCONSTANTS = {
```

**Review by oSumAtrIX — CHANGES_REQUESTED** [review]

File: app/src/main/java/app/revanced/integrations/patches/LithoThemePatch.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        -98492127 // video chapters list background
+    };
+
+    // Used by app.revanced.patches.youtube.layout.theme.patch.LithoThemePatch
+    public static int applyLithoTheme(int originalValue) {
+        if ((!ThemeHelper.isDarkTheme() && anyEquals(originalValue, WHITECONSTANTS)) ||
+            (ThemeHelper.isDarkTheme() && anyEquals(originalValue, DARKCONSTANTS)))
+                return 0;
+        return originalValue;
+    }
```
Comment (oSumAtrIX) [inline app/src/main/java/app/revanced/integrations/patches/LithoThemePatch.java:28]:
```suggestion
    public static int applyLithoTheme(int originalValue) {
        var isDarkTheme = ThemeHelper.isDarkTheme();
        
        if ((isDarkTheme  && anyEquals(originalValue, DARKCONSTANTS)) || (!isDarkTheme  && anyEquals(originalValue, WHITECONSTANTS)))
                return 0;
        return originalValue;
    }
```

**Review by oSumAtrIX — CHANGES_REQUESTED** [review]

**Pull request merged**